src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -100,6 +90,62 @@ class wrap {
     }
 
 private:
+    // Authenticates and authorizes the request when HTTP Basic Auth is enabled
+    // and handles audit logging. It throws on failure.
+    void handle_auth(server::request_t& rq) const {
+        rq.authn_method = config::get_authn_method(
+          rq.service().config().schema_registry_api.value(),
+          rq.req->get_listener_idx());
+
+        if (rq.authn_method != config::rest_authn_method::none) {
+            // Will throw 400 & 401 if auth fails
+            auto auth_result = [this, &rq]() {
+                try {
+                    return rq.service().authenticator().authenticate(*rq.req);
+                } catch (const unauthorized_user_exception& e) {
+                    audit_authn_failure(rq, e.get_username(), e.what());
+                    throw;
+                } catch (const ss::httpd::base_exception& e) {
+                    audit_authn_failure(rq, "", e.what());
+                    throw;
+                }
+            }();
+
+            rq.user = credential_t{
+              auth_result.get_username(),
+              auth_result.get_password(),
+              auth_result.get_sasl_mechanism()};
+            audit_authn_success(rq);
+
+            // Will throw 403 if user enabled HTTP Basic Auth but
+            // did not give the authorization header.
+            [this, &rq, &auth_result]() {
+                try {
+                    switch (_auth_level) {
+                    case auth_level::superuser:
+                        auth_result.require_superuser();
+                        break;
+                    case auth_level::user:
+                        auth_result.require_authenticated();
+                        break;
+                    case auth_level::publik:
+                        auth_result.pass();
+                        break;
+                    }
+                } catch (const ss::httpd::base_exception& e) {
+                    audit_authz_failure(rq, auth_result, e.what());
+                    throw;
+                }
+            }();
```

Comment:
why is this an IIL? it's not being used to initialize a variable, which is the most common use case.